packages/plugin-commands-publishing/test/gitChecks.ts

```diff
@@ -0,0 +1,117 @@
+import PnpmError from '@pnpm/error'
+import { publish } from '@pnpm/plugin-commands-publishing'
+import prepare from '@pnpm/prepare'
+import { REGISTRY_MOCK_PORT } from '@pnpm/registry-mock'
+import execa = require('execa')
+import fs = require('mz/fs')
+import path = require('path')
+import test = require('tape')
+import tempy = require('tempy')
+import { DEFAULT_OPTS } from './utils'
+
+const CREDENTIALS = [
+  `--registry=http://localhost:${REGISTRY_MOCK_PORT}/`,
+  `--//localhost:${REGISTRY_MOCK_PORT}/:username=username`,
+  `--//localhost:${REGISTRY_MOCK_PORT}/:_password=${Buffer.from('password').toString('base64')}`,
+  `--//localhost:${REGISTRY_MOCK_PORT}/:email=foo@bar.net`,
+]
+
+test('publish: fails git check if branch is not on master', async (t) => {
+  const tempDir = tempy.directory()
+
+  prepare(t, {
+    name: 'test-publish-package.json',
+    version: '0.0.0',
+  }, {
+    tempDir,
```

Comment:
This is actually not needed because prepare() will use tempy to create the directory and will change the cwd to that temp folder:

https://github.com/pnpm/pnpm/blob/1b02990a2812b6096836856e5d057ea9589c7fdb/privatePackages/prepare/src/index.ts#L62-L84